### ISSUE TITLE
how to get for windows?

### ISSUE THREAD

**Issue by tjw0lf** [flightlessmango/MangoHud#222]
https://www.reddit.com/r/linux_gaming/comments/h8p9l6/red_dead_redemption_2_benchmark_wine_vs_windows/

I see you have windows vulkan with it how did you do that?

**Comment by flightlessmango** [comment]
This is still in early development, It might not work as intended. GPU load only works on Nvidia so far and many other features are disabled by default

[mangohud.zip](https://flightlessmango.com/mangohud.zip) extract this folder to `C:\`
Run the `reg` file in the mangohud folder
Add `MANGOHUD` as an environment variable with value `1`
Test with `vkcube` that's included.

For dx9 and dx11 you can inject the dll manually.
Open up PowerShell / cmd
Navigate to `C:\mangohud`
for 64bit `.\DLLInjector64.exe -l C:\mangohud\MangoHud.dll -p pid`
for 32bit `.\DLLInjector32.exe -l C:\mangohud\MangoHud32.dll -p pid`

**Comment by tjw0lf** [comment]
I have AMD RX 590. will this work with that

**Comment by flightlessmango** [comment]
Yes it should work fine, it's just that the GPU load will show 0%

**Comment by tjw0lf** [comment]
Where did you play Red Dead 2. Epic or Steam

**Comment by flightlessmango** [comment]
I used steam version

**Comment by Recapi** [comment]
works fine with vulkan cube , https://imgur.com/a/fB5kqrB , but cant inject dll to a game pid

**Comment by NoFuchsGavin** [comment]
![image](https://user-images.githubusercontent.com/26583027/86889932-d79e7880-c0fc-11ea-9cd6-0cb375a0350d.png)

It does work on the _vkcube.exe_ , but I seems to not attach to the Wow.exe as it is described.

**Comment by flightlessmango** [comment]
@NoFuchsGavin what api are you using in wow?

**Comment by NoFuchsGavin** [comment]
> @NoFuchsGavin what api are you using in wow?

I don't understand your question. I am just starting World of Warcraft via Battle.NET Launcher and tried to attach Mangohud like described above to the process.

**Comment by flightlessmango** [comment]
are you using dx11 or dx12 in wow?

**Comment by NoFuchsGavin** [comment]
> are you using dx11 or dx12 in wow?

I tested it with:

- DirectX12
- DirectX11
- DirectX11 Legacy 

Restarted always the game to be sure.

**Comment by SharpOB** [comment]
Using an RX 580, having used the .reg file and launching vkcube with the following batch script:
`MANGOHUD=1 
set MANGOHUD=1
start vkcube.exe`
I encounter the following error: 
![image](https://user-images.githubusercontent.com/24550634/86967048-fbe07080-c161-11ea-852c-bdeba957abba.png)
The mangohud UI appears momentarily in the vkcube window but does not work:
![image](https://user-images.githubusercontent.com/24550634/86967077-0b5fb980-c162-11ea-9364-c6bec926898b.png)
and the application quits after a second.
With the environment variable set I tried to inject the .DLL into GTA IV, to no effect.
As a big fan of the look and utility of mangohud and how light it is, Windows support would be fantastic for me.

**Comment by flightlessmango** [comment]
@SharpOB 
this is an updated version that *should* work with nvidia and amd, however it's not fully tested.
[MangoHud_64.zip](http://flightlessmango.com/MangoHud_64.zip)

**Comment by ZerataX** [comment]
I also fail to inject it into dx9/dx11/dx12/opengl games, but anything that uses vulkan just works via setting the env variable.

**Comment by SharpOB** [comment]
![image](https://user-images.githubusercontent.com/24550634/87330601-ba204300-c530-11ea-9b5f-3cd56ba52485.png)
AMD seems to work now, though injecting into GTA IV (32bit) still seems to do nothing.

**Comment by CloudHaze** [comment]
So you are Mango Rolo (bad boys 3, joke) xD
Cheers from Venezuela, I was looking for some alternatives to Windows Rivastatistic turner, cause i wanted to have like more profiles, sometimes I just ran like in riva 12 cpu +gpu, and sometimes I only wan to measure the frametime, and so far in my knowledge rivastatic has no profiles. so thats why i wanted to use a second alternative to measure only like GPU+cpu+frametime

So does this work for Windows right? and if using Nvidia is all fine? 

I apologies for my poor grammar.

**Comment by flightlessmango** [comment]
@CloudHaze I would say it barely works. This is still in very early development

**Comment by RiantAnda** [comment]
It's working for me with Wolfenstein: Youngblood, which is Vulkan native, however the dll injection method doesn't work for any game I tried.

I've tried so far:
Metro Exodus (DX11)
Ass. Creed Odyssey (?DX11?)
Beyond a Steel Sky (?DX11?)
Horizon Zero Dawn (?DX12?)

What I'm doing:
Launch the game
Get the *game*.exe PID from Task Manager
Run ".\DLLInjector64.exe -l *full path to folder*/MangoHud.dll -p *pid*" from MangoHud folder

@flightlessmango 
Is there any way to verify/debug the dll injection or am I doing something wrong?
What games should work from your side?

**Comment by Doctorwho1909** [comment]
any chance this will work on opengl ? alot of emulators like ryujinx use opengl still

**Comment by ycanales** [comment]
mangohud is awesome, I wish it worked with Windows like RTSS.

**Comment by PrincessAkira** [comment]
Just tried it on an OpenGL Game and also didnt like it 
![image](https://user-images.githubusercontent.com/45071533/101635452-4a07e180-3a2a-11eb-9964-d5bc4f5b1c1a.png)
